include/boost/math/tools/polynomial.hpp

```diff
@@ -134,7 +236,11 @@ class polynomial
 
    // access:
    size_type size()const { return m_data.size(); }
-   size_type degree()const { return m_data.size() - 1; }
+   size_type degree()const
+   { 
+       BOOST_ASSERT(size() != 0);
```

Comment:
But that's the problem, the degree of the zero polynomial is not zero, it's undefined (http://mathworld.wolfram.com/ZeroPolynomial.html). By convention, some systems return -1 or -∞ but neither of those options are easy here. I thought an undefined value translated well to saying that this function is undefined when `size() == 0`. What about throwing an exception? Either way, I don't think users should expect `degree()` to return a useful value for the zero polynomial.